test/src/main/scala/scala/test/twitter_scrooge/BUILD

```diff
@@ -0,0 +1,100 @@
+load("//scala:scala.bzl",   "scala_binary", "scala_library")
+load("//twitter_scrooge:twitter_scrooge.bzl", "scrooge_scala_library")
+
+scrooge_scala_library(
+  name = "thrift1",
+  thrift_library = "//test/src/main/scala/scala/test/twitter_scrooge/thrift",
+  visibility = ["//visibility:public"],
+)
+
+scrooge_scala_library(
```

Comment:
I worry about this:
```
thriftA -> thriftB
scroogeA -> thriftA (will generate thriftB?)
scoogeB -> thriftB
scala_lib -> scroogeA, scroogeB (now we have the scala from thriftB on the path twice, right?)
```
what is the right way to tackle this?

My thinking was that scrooge_scala_srcjar would be non-transitive. It would only generate a srcjar containing files from the given target, but it would depend on a srcjar target for all the deps to the src jar. What do you think?

I think this requires walking backwards on the graph (seeing the scrooge_scala_srcjar that depends on the thrift, and there should be exactly 1) or using the assumption of a standard naming as you have done here.

The last challenge is how to remove the files we don't want. I think we use the mapfile to do that: we only keep scala files that are associated with the direct dependencies in this rule.

What do you think?